src/test/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReferenceTest.java

```diff
@@ -1184,5 +1184,27 @@ void m() {
         );
     }
 
+    @Test
+    void orElseGet() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import java.util.Optional;
+              import java.util.function.Function;
+
+              class A {
+                public void testCase() {
+                  Function<String, Integer> function = str -> 1;
+                  Optional.of(function).orElseGet(() -> this::foo);
```

Comment:
So after applying the recipe we incorrectly end up with
```diff
-Optional.of(function).orElseGet(() -> this::foo);
+Optional.of(function).orElseGet(this::foo);
```
Whereas we should not have made a change if I understand you correctly.